OpenAPI/kiota/tutorials/dotnet-dependencyinjection.md

```diff
@@ -0,0 +1,288 @@
+---
+title: Use Kiota with dependency injection in .NET
+description: Learn how you can combine Kiota with dependency injection in .NET to build API clients.
+author: svrooij
+ms.author: vibiret
+ms.topic: tutorial
+date: 05/10/2024
+---
+
+# Register a Kiota API client in .NET with dependency injection
+
+In this tutorial, you will learn how to register a [Kiota API client](/openapi/kiota/tutorials/dotnet-azure?wt.mc_id=SEC-MVP-5004985) with [dependency injection](/dotnet/core/extensions/dependency-injection-usage?wt.mc_id=SEC-MVP-5004985).
+
+## Required tools
+
+- [.NET SDK 8.0](https://get.dot.net/8)
+- [Kiota CLI](/openapi/kiota/install?tabs=bash&wt.mc_id=SEC-MVP-5004985#install-as-net-tool)
+
+## Create a project
+
+Run the following command in the directory you want to create a new project.
+
+```dotnetcli
+dotnet new webapi -o kiota-with-di
```

Comment:
add webapi to this file please https://github.com/MicrosoftDocs/openapi-docs/blob/main/cspell.json